### ISSUE TITLE
Significant Number of Unphased Reads

### ISSUE THREAD

**Issue by iek** [whatshap/whatshap#358]
Hi Marcel,

I'm sorry for the additional issue, but I just wanted to ask if it was normal to have a significant number of unphased reads (specifically 99.99% of reads were unphased). I installed whatshap via conda and am using the developers' version you recently uploaded along with Python 3.10 . The command I ran was as follows:

`whatshap phase -o phased_YB5001.vcf --reference=isaac_Pf3D7.fasta --ignore-read-groups --indels var_freebayes_YB5001_illumina.vcf aln_nanopore_YB5001.sorted.bam`

This was the final stats after running Whatshap:

`---------------- ALL chromosomes (aggregated) ----------------
     Variants in VCF: 23037643
        Heterozygous: 23031189 (    3342 SNVs)
              Phased:     3161 (    2767 SNVs)
            Unphased: 23028026 (not considered below)
          Singletons:        2 (not considered below)
              Blocks:      390

Block sizes (no. of variants)
   Median block size:        3.00 variants
  Average block size:        8.11 variants
       Largest block:      119    variants
      Smallest block:        2    variants

Block lengths (basepairs)
      Sum of lengths:   936654    bp
 Median block length:      921.00 bp
Average block length:     2401.68 bp
       Longest block:    27083    bp
      Shortest block:        3    bp
          Block NG50:        0    bp`

If this is not normal, do you have any ideas as to what caused 99.99% of the reads to be unphased? Thank you so much.

**Comment by marcelm** [comment]
I will probably not have time today, but I assuming this is the data you already sent me, I will have a look.

**Comment by iek** [comment]
Absolutely. Thank you so much. The data is essentially the same except for the minor manual modifications I made to the VCF from FreeBayes. I have sent you the updated files. Thank you again.

**Comment by marcelm** [comment]
You sent me a file named `YB5001_long_reads_raw_aln.sorted.bam`, but the phasing command you give above uses the file name `aln_nanopore_YB5001.sorted.bam`. Are these the same files?

**Comment by iek** [comment]
Hi Marcel,

Thank you for your quick response. They should be the same files, but I have sent you the new file (aln_nanopore_YB5001.sorted.bam) nevertheless. Thank you so much!

**Comment by marcelm** [comment]
(The files were not identical (different contig order), but it doesn’t matter.)

I apologize if I explain something that you already know quite well, but just in case: Your VCF file is a gVCF ("genome VCF"). That is, it includes records for sites that do not represent variants. In the Freebayes command that you gave above, you did not include the `--gvcf` option, but the header says that it was generated with `--gvcf --gvcf-dont-use-chunk true` (and you listed that option in an earlier e-mail to me). The `--gvcf-dont-use-chunk true` option causes Freebayes to create a VCF record for *each position* of the reference. Since your reference has a total length of around 23 million nucleotides, there are also around 23 million records in the VCF.

When `whatshap stats` is used on gVCF files, it erroneously includes these non-variants when counting variants. This is why it reports such a high total number of "Variants in VCF" of 23037643.

`whatshap stats` should be changed to ignore the non-variant positions, and I have opened #359 for that. I won’t have time to make this change now, so here are two things you can do to get accurate numbers:

- Run Freebayes without the `--gvcf` options
- Remove all records from the VCF that do not have a `GT` field. I have used this command: `egrep '(^#|GT:)' phased_YB5001.vcf > phased_YB5001.gt.vcf`

The advantage is that these files are much smaller so that `whatshap stats` runs much faster.

**Comment by iek** [comment]
Hi Marcel,

Thank you very much for your thorough explanation. Does that mean that Whatshap Stats is incorrect but Whatshap Phase and Haplotag run correctly with the gVCF files? I did visualize the haplotagged reads on IGV, and it seemed that very few reads were assigned a haplotype (I’m not sure it’s quite .01% but it is definitely less than 1% of reads).

**Comment by marcelm** [comment]
Both `whatshap phase` and `haplotag` should work correctly in the developer version that you have. `whatshap stats` is incorrect, yes.

Your initial guess that you’re too strict in calling variants could be right. Looking at your data in IGV, it seems that the called variants are quite sparse (reference length is 23 million nt, you have 11000 variants in total, so distance between each variant is ~2000 nt on average). Your reads have an average length of 3427 nt, but there is a lot of soft-clipping, so on average only 1261 bp of each read are aligned to the reference. That means that many of them will fall in between variants and cannot be haplotagged.

I don’t think there’s much more I can do from the WhatsHap side.

**Comment by charliechen912ilovbash** [comment]
@marcelm Hi, I am also encountering similar situation. There are some reads containing soft-clipped nucleotides due to translocation between chromosomes in my sample. There is ~60%  reads untagged, which is higher than I expected. 
After observing the H1, H2 and untagged read in IGV, I have some questions:
1. The read will be categorized into untagged if it is shorter than how many bp?
2. How does whatshap deal with soft-clipped bases ?
3. I observed some reads are untagged probably because there is no enough density of heterozygous SNV,  is 1 SNV per 1000 bp enough for phasing a read?
4. Other possible reason of untagged read, except low mapping quality, short read length and low density of heterozygous SNV
Thank you very much.
